src/pybind/mgr/devicehealth/module.py

```diff
@@ -399,31 +399,27 @@ def show_device_metrics(self, devid, sample):
             return -errno.ENOENT, '', 'device ' + devid + ' not found'
         # fetch metrics
         res = {}
-        ioctx = self.open_connection(create_if_missing=False)
-        if ioctx:
-            try:
-                with rados.ReadOpCtx() as op:
-                    omap_iter, ret = ioctx.get_omap_vals(op, "", sample or '', 500)  # fixme
-                    assert ret == 0
-                    try:
-                        ioctx.operate_read_op(op, devid)
-                        for key, value in list(omap_iter):
-                            if sample and key != sample:
-                                break
-                            try:
-                                v = json.loads(value)
-                            except (ValueError, IndexError):
-                                self.log.debug('unable to parse value for %s: "%s"' %
-                                               (key, value))
-                                pass
-                            res[key] = v
-                    except rados.ObjectNotFound:
-                        pass
-                    except rados.Error as e:
-                        self.log.exception("RADOS error reading omap: {0}".format(e))
-                        raise
-            finally:
-                ioctx.close()
+        with self.open_connection(create_if_missing=False) as ioctx:
+            with rados.ReadOpCtx() as op:
+                omap_iter, ret = ioctx.get_omap_vals(op, "", sample or '', 500)  # fixme
+                assert ret == 0
+                try:
+                    ioctx.operate_read_op(op, devid)
+                    for key, value in list(omap_iter):
+                        if sample and key != sample:
+                            break
+                        try:
+                            v = json.loads(value)
+                        except (ValueError, IndexError):
+                            self.log.debug('unable to parse value for %s: "%s"' %
+                                           (key, value))
+                            pass
```

Comment:
[OT] why `pass` here? if `json.loads()` throws, the `v` will be the one parsed in previous iteration, while `key` is the current one.